lib/daru/category.rb

```diff
@@ -0,0 +1,883 @@
+    #   #   b   2
+    #   #   c   3
+    def reindex! idx
+      idx = Daru::Index.new idx unless idx.is_a? Daru::Index
+      raise ArgumentError, 'Invalid index specified' unless
+        idx.to_a.sort == index.to_a.sort
+
+      old_categories = categories
+      data = idx.map { |i| self[i] }
+      initialize_core_attributes data
+      self.categories = old_categories
+      self.index = idx
+      self
+    end
+
+    # Non-destructive version of #reorder!
+    # See #reorder!
+    def reorder order
+      dup.reorder! order
+    end
+
+    # Non-destructive version of #reindex!
+    # See #reindex!
+    def reindex index
+      dup.reindex! index
+    end
+
+    {
+      eq: :==,
+      not_eq: :!=,
+      lt: :<,
+      lteq: :<=,
+      mt: :>,
+      mteq: :>=
+    }.each do |method, operator|
+      define_method(method) do |other|
+        mod = Daru::Core::Query
+        if other.is_a?(Daru::Vector)
+          mod.apply_vector_operator operator, to_ints, other.to_ints
+        else
+          mod.apply_scalar_operator operator, @array, int_from_cat(other)
+        end
+      end
+    end
+    alias :gt :mt
+    alias :gteq :mteq
+
+    # For querying the data
+    # @param [object] arel like query syntax
+    # @return [Daru::Vector] Vector which makes the conditions true
+    # @example
+    #   dv = Daru::Vector.new ['I', 'II', 'I', 'III', 'I', 'II'],
+    #     type: :category,
+    #     ordered: true,
+    #     categories: ['I', 'II', 'III']
+    #   dv.where(dv.mt('I') & dv.lt('III'))
+    #   # => #<Daru::Vector(2)>
+    #   #   1  II
+    #   #   5  II
+    def where bool_array
+      Daru::Core::Query.vector_where self, bool_array
+    end
+
+    # Gives the summary of data using following parameters
+    # - size: size of the data
+    # - categories: total number of categories
+    # - max_freq: Max no of times a category occurs
+    # - max_category: The category which occurs max no of times
+    # - min_freq: Min no of times a category occurs
+    # - min_category: The category which occurs min no of times
+    # @return [Daru::Vector] Vector with index as following parameters
+    #   and values as values to these parameters
+    # @example
+    #   dv = Daru::Vector.new [:a, 1, :a, 1, :c], type: :category
+    #   dv.summary
+    #   # => #<Daru::Vector(6)>
+    #   #         size            5
+    #   #   categories            3
+    #   #     max_freq            2
+    #   # max_category            a
+    #   #     min_freq            1
+    #   # min_category            c
+    def summary
+      values = {
+        size: size,
+        categories: categories.size,
+        max_freq: @cat_hash.values.map(&:size).max,
+        max_category: @cat_hash.keys.max_by { |cat| @cat_hash[cat].size },
+        min_freq: @cat_hash.values.map(&:size).min,
+        min_category: @cat_hash.keys.min_by { |cat| @cat_hash[cat].size }
+      }
+
+      Daru::Vector.new values
+    end
+
+    alias_method :describe, :summary
```

Comment:
You are right. I have renamed this function to `describe` and removed the name `summary`. It can be used for some other function in the future.